DROD/Main.cpp

```diff
@@ -1455,7 +1455,9 @@ bool IsAppAlreadyRunning()
 	tmp += dotpid;
 	const UINT lflen = tmp.length();
 	if (!(lockfile = new char[lflen + 1])) return true;
-	UnicodeToUTF8(tmp, lockfile);
+	string lockfileString;
```

Comment:
Thanks for this fix.  I agree this will work.  However, we should generally shy away from using char[] buffers for string management in the codebase.  Whenever one of these causes us an issue, we should probably refactor it to be a std::string so we don't need to include code for the memory management, asserting it's non-null wherever it's used, and so forth.  I'm certain I'm to blame for introducing many of these, but as I look back, I acknowledge they're not optimal and we should replace their usage.

Reply:
Hmm yes that sounds logical, but in this case a lot of file-system functions are called which all require a char[] buffer. Doesn't it make more sense to leave it as is in this specific case?

Reply:
Good question.  What I'm recommending to do here (and elsewhere, where we allocate char[] buffers) is remove the `lockfile` variable, remove
`strcpy(lockfile, lockfileString.c_str());`
and replace `FILE *fp = fopen(lockfile, "r");` with `FILE *fp = fopen(lockfileString.c_str(), "r");`
This makes more sense to me in order to avoid the superfluous char buffer allocation, since the buffer is already internally allocated inside the std::string var.

Reply:
I'm going to approve and merge this PR and apply my recommended improvements on my end.